modules/instance_template/main.tf

```diff
@@ -18,6 +18,13 @@
 # Locals
 #########
 
+data "google_service_account" "existing_sa" {
+  provider   = google-beta
+  count      = local.create_service_account == false ? 1 : 0
+  account_id = element(split("@", google_compute_instance_template.tpl.service_account[0].email), 0)
```

Comment:
I believe just email can be directly passed per https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/service_account#account_id-2